### PR TITLE
Single API -- Upload image with tags

### PR DESCRIPTION
**Pull request by sudoDeznit**
**Description**
- Use new unified api --> `/api/photos/submit-with-tags` to upload images
- Send `picked_up` property on api `/api/photos/submit-with-tags` & `/api/add-tags`
- Move uploadImage, uploadTags, deleteWebImage, checkWebImage to `image_action` because it's not a shared action there is only one images array and `image_reducer` is used after refactor.
- Refactor `checkForImagesOnWeb` and `deleteWebImage` to use async/await .

**Trello Card**
[Upload Image + Tags in a single request](https://trello.com/c/ShmE6VDr)

**Note**
- This PR needs to be merged after #127 .

### CODE REVIEW

File: reducers/images_reducer.js [hunk 1]
```diff
@@ -56,7 +57,7 @@ export default function(state = INITIAL_STATE, action) {
                                     type: action.payload.type,
                                     selected: false,
                                     tags: {},
-                                    picked_up: false,
+                                    pickedUp: action.payload.pickedUp,
```
Comment (xlcrr) [inline reducers/images_reducer.js:60]: keep `picked_up` everywhere

File: screens/home/HomeScreen.js [hunk 2]
```diff
@@ -394,13 +393,18 @@ class HomeScreen extends PureComponent {
...
                     img.tags &&
                     Object.keys(img.tags).length > 0 &&
                     isgeotagged
                 ) {
+                    const date =
+                        img.type === 'GALLERY'
+                            ? moment
+                                  .unix(img.date)
+                                  .format('YYYY:MM:DD HH:mm:ss')
+                            : img.date;
```
Comment (xlcrr) [inline screens/home/HomeScreen.js:407]: why don't we do this when the image is selected?

File: screens/home/homeComponents/UploadImagesGrid.js [hunk 3]
```diff
@@ -11,7 +11,6 @@ const { width: SCREEN_WIDTH, height: SCREEN_HEIGHT } = Dimensions.get('window');
 class UploadImagesGrid extends PureComponent {
     imagePressed(index) {
         const image = this.props.photos[index];
-
```
Comment (xlcrr) [inline screens/home/homeComponents/UploadImagesGrid.js:14]: don't delete this

**Review by xlcrr — CHANGES_REQUESTED** [review]

File: actions/images_action.js [hunk 4]
```diff
@@ -110,9 +182,97 @@ export const toggleSelecting = () => {
...
+ * fn to upload images along with tags
+ * @param {string} token
+ * @param  image form data
+ * @returns
+ */
+export const uploadImage = (token, image, imageId) => {
+    let response;
+    return async (dispatch) => {
+        try {
+            response = await axios(URL + '/api/photos/submit-with-tags', {
```
[the remark this answers is not in the document]
Comment (sudoDeznit) [reply actions/images_action.js:202]: According to this [backend PR description](https://github.com/OpenLitterMap/openlittermap-web/pull/364#issue-1031510879) endpoint is `/api/photos/submit-with-tags`

File: actions/images_action.js [hunk 5]
```diff
@@ -17,25 +19,65 @@ import {
...
+            } else {
+                console.log('checkForImagesOnWeb -- Network Error');
+            }
+        }
+        if (response && response?.data?.photos) {
+            let photos = response.data.photos;
+            dispatch({
+                type: ADD_IMAGE,
+                payload: {
+                    images: photos,
```
[the remark this answers is not in the document]
Comment (sudoDeznit) [reply actions/images_action.js:72]: yes, array of photos

File: reducers/images_reducer.js (hunk 1 above)
Comment (sudoDeznit) [reply reducers/images_reducer.js:60]: we have used `camelCase` for all the variables declared on frontend.

File: screens/home/HomeScreen.js (hunk 2 above)
Comment (sudoDeznit) [reply screens/home/HomeScreen.js:407]: We can do that when we select image from Galley... better to do that there, I will move this

File: actions/images_action.js [hunk 6]
```diff
@@ -17,25 +19,65 @@ import {
...
 };
 
+/**
+ * Get images uploaded from website but not yet tagged
+ *
+ * @param token - jwt
+ * @param pickedUp - default user setting if litter is pickedUp or not
+ *
+ */
+export const checkForImagesOnWeb = (token, pickedUp) => {
```
[the remark this answers is not in the document]
Comment (sudoDeznit) [reply actions/images_action.js:48]: we can use `getState` of redux thunk to get value of `picked_up` , but its better to have pass variable than using `getState`

File: screens/home/HomeScreen.js (hunk 2 above)
Comment (sudoDeznit) [reply screens/home/HomeScreen.js]: Current backend can parse epoch timestamp.. So sending epoch now, and removed this date formatting on frontend e619afa4739ede4b26d03a8717c2de93bdf5610d

File: actions/images_action.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (xlcrr) [reply actions/images_action.js:202]: Yes, but submit doesn't make much sense, correct word should be `upload`. Small detail, not urgent

File: actions/images_action.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (xlcrr) [reply actions/images_action.js:72]: should it be ADD_IMAGES?

**Comment by xlcrr** [comment]
@sudoDeznit please prepare this for merge

File: actions/images_action.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (sudoDeznit) [reply actions/images_action.js:72]: changed to ADD_IMAGES

File: actions/images_action.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (sudoDeznit) [reply actions/images_action.js]: Changed to `upload-with-tags` based in [backend PR](https://github.com/OpenLitterMap/openlittermap-web/pull/373/commits/db6d48740e5032485f70a9776885f8c9d8da1d72)

**Comment by sudoDeznit** [comment]
fixed merge conflicts, changed `submit-with-tags` to `upload-with-tags` and `ADD_IMAGE` to `ADD_IMAGES` and merged to staging

**Pull request merged**